server/runtime_nakama_module.go

```diff
@@ -0,0 +1,1108 @@
+				return
+			}
+			stream.Subject = did
+		case "Label":
+			if v.Type() != lua.LTString {
+				conversionError = "stream Label must be a string"
+				return
+			}
+			stream.Label = v.String()
+		}
+	})
+	if conversionError != "" {
+		l.ArgError(3, conversionError)
+		return 0
+	}
+
+	meta := n.tracker.GetLocalBySessionIDStreamUserID(sessionID, stream, userID)
+	if meta == nil {
+		l.Push(lua.LNil)
+	} else {
+		metaTable := l.NewTable()
+		metaTable.RawSetString("Hidden", lua.LBool(meta.Hidden))
+		metaTable.RawSetString("Persistence", lua.LBool(meta.Persistence))
+		metaTable.RawSetString("Username", lua.LString(meta.Username))
+		metaTable.RawSetString("Status", lua.LString(meta.Status))
+		l.Push(metaTable)
+	}
+	return 1
+}
+
+func (n *NakamaModule) streamUserJoin(l *lua.LState) int {
+	// Parse input User ID.
+	userIDString := l.CheckString(1)
+	if userIDString == "" {
+		l.ArgError(1, "expects user id")
+		return 0
+	}
+	userID, err := uuid.FromString(userIDString)
+	if err != nil {
+		l.ArgError(1, "expects valid user id")
+		return 0
+	}
+
+	// Parse input Session ID.
+	sessionIDString := l.CheckString(2)
+	if sessionIDString == "" {
+		l.ArgError(2, "expects session id")
+		return 0
+	}
+	sessionID, err := uuid.FromString(sessionIDString)
+	if err != nil {
+		l.ArgError(2, "expects valid session id")
+		return 0
+	}
+
+	// Parse input stream identifier.
+	streamTable := l.CheckTable(3)
+	if streamTable == nil || streamTable.Len() == 0 {
+		l.ArgError(3, "expects a valid stream")
+		return 0
+	}
+	stream := PresenceStream{}
+	conversionError := ""
+	streamTable.ForEach(func(k lua.LValue, v lua.LValue) {
+		switch k.String() {
+		case "Mode":
+			if v.Type() != lua.LTNumber {
+				conversionError = "stream Mode must be a number"
+				return
+			}
+			stream.Mode = uint8(lua.LVAsNumber(v))
+		case "Subject":
+			if v.Type() != lua.LTString {
+				conversionError = "stream Subject must be a string"
+				return
+			}
+			sid, err := uuid.FromString(v.String())
+			if err != nil {
+				conversionError = "stream Subject must be a valid identifier"
+				return
+			}
+			stream.Subject = sid
+		case "Descriptor":
+			if v.Type() != lua.LTString {
+				conversionError = "stream Descriptor must be a string"
+				return
+			}
+			did, err := uuid.FromString(v.String())
+			if err != nil {
+				conversionError = "stream Descriptor must be a valid identifier"
+				return
+			}
+			stream.Subject = did
+		case "Label":
+			if v.Type() != lua.LTString {
+				conversionError = "stream Label must be a string"
+				return
+			}
+			stream.Label = v.String()
+		}
+	})
+	if conversionError != "" {
+		l.ArgError(3, conversionError)
+		return 0
+	}
+
+	// By default generate presence events.
+	hidden := l.OptBool(4, false)
+	// By default persistence is enabled, if the stream supports it.
+	persistence := l.OptBool(5, true)
+
+	// Look up the session.
+	session := n.registry.Get(sessionID)
```

Comment:
This only looks up the session on the local node, not cluster wide. I think we should have an offline call to figure out what the best thing here is.

Reply:
As above, stream presences are checked/mutated on the node they belong to only, so this is correct behaviour until that changes.